src/strategies/Primitives/LimitSwapExactInput.ts

```diff
@@ -0,0 +1,115 @@
+import { TokenArgs, PrimitiveParamType, BigIntish } from '@brinkninja/types'
+import Token from '../Token'
+import InputTokenPrimitive from './InputTokenPrimitive'
+
+export type LimitSwapExactInputArgs = {
+  priceCurveAddress: string
+  priceCurveParams: string
+  signer: string
+  tokenIn: TokenArgs
+  tokenOut: TokenArgs
+  tokenInAmount: BigIntish,
+  fillStateParams: FillStateParamsJSON
+}
+
+export type FillStateParamsJSON = {
+  id: BigIntish
+  startX96: BigIntish
+  sign: boolean
+}
+
+export const LimitSwapExactInputFunctionParams: PrimitiveParamType[] = [
+  {
+    name: 'priceCurveAddress',
+    type: 'address',
+    signed: true
+  },
+  {
+    name: 'priceCurveParams',
+    type: 'bytes',
+    signed: true
+  },
+  {
+    name: 'owner',
+    type: 'address',
+    signed: true
+  },
+  {
+    name: 'tokenIn',
+    type: 'Token',
+    signed: true
+  },
+  {
+    name: 'tokenOut',
+    type: 'Token',
+    signed: true
+  },
+  {
+    name: 'tokenInAmount',
+    type: 'uint256',
+    signed: true
+  },
+  {
```

Comment:
since these need to map the exact input params of limitSwapExactInput fn, these fill state params shouldn't be split. should just be one param for the struct:

```
{
  name: 'fillStateParams',
  type: 'FillStateParams',
  signed: true
}
```